pkg/flow/flow.go

```diff
@@ -209,9 +214,66 @@ func (c *Flow) LoadFile(f *File) error {
 	return diags.ErrorOrNil()
 }
 
+// ComponentInfos returns the component infos.
+func (c *Flow) ComponentInfos() []*river.ComponentField {
+	cns := c.loader.Components()
+	infos := make([]*river.ComponentField, len(cns))
+	refByBacktrack := make(map[string]*river.ComponentField, 0)
+	for i, com := range cns {
```

Comment:
You should be able to look at the edges of the graph directly instead of having to call `ComponentReferences` (which isn't cheap) or having to construct the backtracking map